File: cplusplus/src/class3.cpp
#include "class3.hpp"

libimport int class3_add(int a, int b, int c, int d);
libimport void class3_hello(void);

int class3::add(int a, int b, int c, int d) {
    return class3_add(a,b,c,d);
}

void class3::hello(void){
    return class3_hello();
}

File: csharp/Class3.cs
using System;
using System.Runtime.InteropServices;

namespace CppCsComTest {

    public class Class3 {
        //export
        [UnmanagedCallersOnly(EntryPoint = "class3_add")]
        public static int Add(int a, int b, int c, int d) {
            return Class2.Add( a, Class1.Add(b,c,d) );
        }
        //export
        [UnmanagedCallersOnly(EntryPoint = "class3_hello")]
        public static void Hello() {
            Class1.Hello();
            Console.WriteLine("Hello again. [C#]");
            return;
        }
    }

}